### ISSUE TITLE
Can we have custom callback for configuration data

### ISSUE THREAD

**Issue by sukesh9995** [CESNET/Netopeer2#1677]
Hi @michalvasko 

module: test
  +--rw leaf1
     +--rw leaf2* [name]
        +--rw name            string
        +--rw value1           string
        +--rw value2           string

Is there any possiblity that when a get-config request comes client, can we register a callback so that we can send the custom data instead of the data stored in running database.

Regards,
Sukesh

**Comment by sukesh9995** [comment]
Hi @michalvasko,

Basically can we get a callback for configuration data, so that we can frame the rpc and reply.

Regards,
Sukesh

**Comment by jktjkt** [comment]
That's typically not how sysrepo is used. What problem are you trying to solve this way, and what exactly are your requirements? Are you maybe syncing configs between multiple backends, etc?

**Comment by michalvasko** [comment]
> Is there any possiblity that when a get-config request comes client, can we register a callback so that we can send the custom data instead of the data stored in running database.

No, this is not possible and should be **avoided**. Such a design is bypassing the design of NMDA and I am guessing replacing it with something similar. So, you should instead understand the difference between `running` and `operational` datastore and then perhaps set some `operational` datastore callback, if required. Read through the [NMDA RFC](https://www.rfc-editor.org/rfc/rfc8342.html) and understand how it is implemented in *sysrepo* by reading its [docs](https://netopeer.liberouter.org/doc/sysrepo/devel/html/index.html#oper_ds).
